src/main/java/bot/Bot.java

```diff
@@ -0,0 +1,218 @@
+package bot;
+
+import bot.enums.Command;
+import bot.exceptions.BotException;
+import bot.exceptions.InvalidCommandException;
+import bot.exceptions.EmptyTodoException;
+import bot.exceptions.InvalidTaskIdException;
+import bot.exceptions.InvalidTaskDescriptionException;
+import bot.storage.Storage;
+import bot.tasks.TaskList;
+import bot.tasks.Todo;
+import bot.tasks.Task;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+public class Bot {
+    private final TaskList tasks;
+    private final Ui ui;
+    private final Parser parser;
+    private final Storage storage;
+
+    public Bot() {
+        tasks = new TaskList();
+        ui = new Ui();
+        parser = new Parser();
+        storage = new Storage();
+        try {
+            storage.loadTasks(tasks);
+        } catch (FileNotFoundException e) {
+            System.out.println("Failed to load task from disk" + e.getMessage());
+        }
+    }
+
+    public void run() {
+        ui.start();
+
+        // TODO: Abstract scanner into Ui class as well
+        Scanner sc = new Scanner(System.in);
+
+        while (sc.hasNextLine()) {
+            handleInput(sc.nextLine());
+        }
+    }
+
+    private void handleInput(String input) {
+        Parser.ParsedInput parsedInput;
+        try {
+            parsedInput = parser.parseInput(input);
+        } catch (InvalidCommandException e) {
+            ui.printMessage(e.getMessage());
+            return;
+        }
+
+        Command cmd = parsedInput.getCmd();
+        String args = parsedInput.getArgs();
+
+        try {
+            switch (cmd) {
+            case LIST:
+                handleList();
+                break;
+            case TODO, DEADLINE, EVENT:
+                handleAddTask(cmd, args);
+                break;
+            case MARK:
+                handleMarkTask(args);
+                break;
+            case UNMARK:
+                handleUnmarkTask(args);
+                break;
+            case DELETE:
+                handleDeleteTask(args);
+                break;
+            case FIND:
+                handleFindTask(args);
+                break;
+            case BYE:
+                ui.exit();
+                System.exit(0);
+                // Fallthrough
+            default:
+                // This should never happen
+                ui.printMessage("Command not found");
+                // Fallthrough
+            }
+        } catch (BotException e) {
+            ui.printMessage(e.getMessage());
+        }
+    }
+
+    private void handleList() {
```

Comment:
Could rename to displayList or displayTaskList to show that it displays, instead of just dealing with a generic list